homeassistant/components/forecast_solar/config_flow.py

```diff
@@ -0,0 +1,138 @@
+"""Config flow for Forecast.Solar integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from forecast_solar import ForecastSolar, ForecastSolarError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_AZIMUTH,
+    CONF_DAMPING,
+    CONF_DECLINATION,
+    CONF_MODULES_POWER,
+    DOMAIN,
+)
+
+
+class ForecastSolarFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Forecast.Solar."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> ForecastSolarOptionFlowHandler:
+        """Get the options flow for this handler."""
+        return ForecastSolarOptionFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            session = async_get_clientsession(self.hass)
+            forecast = ForecastSolar(
+                azimuth=(user_input[CONF_AZIMUTH] - 180),
+                declination=user_input[CONF_DECLINATION],
+                kwp=(user_input[CONF_MODULES_POWER] / 1000),
+                latitude=user_input[CONF_LATITUDE],
+                longitude=user_input[CONF_LONGITUDE],
+                session=session,
+            )
+
+            try:
+                await forecast.estimate()
+            except ForecastSolarError:
+                errors["base"] = "cannot_connect"
```

Comment:
Do we really need to test this ? It's a cloud API so we can just assume it's ok?

Reply:
Hmmm this is not a connection error actually. I'll adjust that to make that more clear.

For example, it could be a bad combination of parameters (although we protect against that).

Reply:
I've removed it. We don't connection check on the options flow either...